src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -148,6 +154,44 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
                                     // Non-public properties should not be included for (de)serialization.
                                 }
                             }
+
+                            foreach (FieldInfo fieldInfo in currentType.GetFields(binding))
```

Comment:
Adding to the property/field cache needs to be dependent on an if-check for `options.IncludeFields`, or the presence of `JsonInclude` on the field.